pkg/controller/core.oam.dev/v1alpha2/application/revision.go

```diff
@@ -45,8 +46,8 @@ func (h *appHandler) GenerateRevision(ctx context.Context, ac *v1alpha2.Applicat
 	appRev := &v1alpha2.ApplicationRevision{
 		Spec: v1alpha2.ApplicationRevisionSpec{
 			Application:              *copiedApp,
-			Components:               comps,
-			ApplicationConfiguration: *ac,
+			Components:               convertComponentList2Map(comps),
+			ApplicationConfiguration: runtime.RawExtension{Object: ac},
```

Comment:
k8s will remove object after persisting so it will make the compare fail, please convert it to raw byte (there is a helper function in the helpers.go) before persiting

Reply:
fixed